client/packages/coldchain/src/Equipment/ImportAsset/EquipmentImportModal.tsx

```diff
@@ -0,0 +1,307 @@
+              ).catch(err => {
+                if (!err) {
+                  err = { message: t('messages.unknown-error') };
+                }
+                importErrorRows.push({
+                  ...asset,
+                  errorMessage: err.message,
+                });
+              });
+            } else {
+              await insertAssets(
+                toInsertEquipmentInput(asset, catalogueItemData?.nodes)
+              ).catch(err => {
+                if (!err) {
+                  err = { message: t('messages.unknown-error') };
+                }
+                importErrorRows.push({
+                  ...asset,
+                  errorMessage: err.message,
+                });
+              });
+            }
+          })
+        ).then(() => {
+          // Update Progress Bar
+          const percentComplete =
+            100 - (remainingRecords.length / numberImportRecords) * 100.0;
+          setImportProgress(percentComplete);
+          setImportErrorCount(importErrorRows.length);
+        });
+      }
+      if (importErrorRows.length === 0) {
+        const importMessage = t('messages.import-generic', {
+          count: numberImportRecords,
+        });
+        const successSnack = success(importMessage);
+        successSnack();
+        onChangeTab(Tabs.Upload);
+        setBufferedEquipment([]);
+        setErrorMessage('');
+        onClose();
+      } else {
+        // Load the error rows in to the component for review
+        setErrorMessage(t('messages.import-error'));
+        setBufferedEquipment(importErrorRows);
+        setImportErrorCount(importErrorRows.length);
+        onChangeTab(Tabs.Review);
+      }
+    }
+  };
+
+  const onClickStep = (tabName: string) => {
+    switch (tabName) {
+      case Tabs.Upload:
+        changeTab(tabName as Tabs);
+        break;
+      case Tabs.Review:
+        changeTab(tabName as Tabs);
+        break;
+      case Tabs.Import:
+        // Do nothing, user can't get to the import page without clicking the import button
+        break;
+    }
+  };
+
+  const changeTab = (tabName: Tabs) => {
+    switch (tabName) {
+      case Tabs.Upload:
+        setErrorMessage('');
+        setBufferedEquipment([]);
+        setActiveStep(0);
+        break;
+      case Tabs.Review:
+        setActiveStep(1);
+        break;
+      case Tabs.Import:
+        setActiveStep(2);
+        break;
+    }
+    onChangeTab(tabName);
+  };
+
+  const importNotReady =
+    bufferedEquipment.length == 0 || errorMessage.length > 0 || isLoading;
+  const exportNotReady = !(
+    bufferedEquipment.length >= 0 && errorMessage.length > 0
+  );
+
+  const importSteps = [
+    { label: t('label.upload'), description: '', clickable: true },
+    { label: t('label.review'), description: '', clickable: true },
+    {
+      label: t('label.import'),
+      description: '',
+      clickable: false,
+    },
+  ];
+
+  return (
+    <Modal
+      okButton={
+        <DialogButton
```

Comment:
can you change this to be `NextButton` when on steps 1 & 2 please.
It would be helpful to have
- next button visible, ok button hidden on step 1&2
- next button hidden, ok visible on step 3

Reply:
I'm just thinking through this case. Currently I'm automatically going to the next stage on upload - but do you think it would be better to wait for the user to click the button?

Currently also uploaded data is cleared when we move back to step one - but maybe should cancel this behaviour also.

I'm a little unclear on what the best behaviour would be

Reply:
I'm wondering if we need a next button at all and in stead automatically move to the next step on upload or on import